doc/classes/SpinBox.xml

```diff
@@ -71,8 +71,98 @@
 		</member>
 	</members>
 	<theme_items>
+		<theme_item name="down_disabled_icon_modulate" data_type="color" type="Color" default="Color(0.875, 0.875, 0.875, 0.5)">
+			Down button icon modulation color, when the button is disabled.
+		</theme_item>
+		<theme_item name="down_hover_icon_modulate" data_type="color" type="Color" default="Color(0.95, 0.95, 0.95, 1)">
+			Down button icon modulation color, when the button is hovered.
+		</theme_item>
+		<theme_item name="down_icon_modulate" data_type="color" type="Color" default="Color(0.875, 0.875, 0.875, 1)">
+			Down button icon modulation color.
+		</theme_item>
+		<theme_item name="down_pressed_icon_modulate" data_type="color" type="Color" default="Color(0.95, 0.95, 0.95, 1)">
+			Down button icon modulation color, when the button is being pressed.
+		</theme_item>
+		<theme_item name="up_disabled_icon_modulate" data_type="color" type="Color" default="Color(0.875, 0.875, 0.875, 0.5)">
+			Up button icon modulation color, when the button is disabled.
+		</theme_item>
+		<theme_item name="up_hover_icon_modulate" data_type="color" type="Color" default="Color(0.95, 0.95, 0.95, 1)">
+			Up button icon modulation color, when the button is hovered.
+		</theme_item>
+		<theme_item name="up_icon_modulate" data_type="color" type="Color" default="Color(0.875, 0.875, 0.875, 1)">
+			Up button icon modulation color.
+		</theme_item>
+		<theme_item name="up_pressed_icon_modulate" data_type="color" type="Color" default="Color(0.95, 0.95, 0.95, 1)">
+			Up button icon modulation color, when the button is being pressed.
+		</theme_item>
+		<theme_item name="buttons_vertical_separation" data_type="constant" type="int" default="0">
+			Vertical separation between the up and down buttons.
+		</theme_item>
+		<theme_item name="buttons_width" data_type="constant" type="int" default="16">
+			Width of the up and down buttons. If smaller than any icon set on the buttons, the said icon may overlap neighboring elements, unless [theme_item set_min_buttons_width_from_icons] is enabled.
+		</theme_item>
+		<theme_item name="field_and_buttons_separation" data_type="constant" type="int" default="2">
+			Width of the horizontal separation between the text input field ([LineEdit]) and the buttons.
+		</theme_item>
+		<theme_item name="set_min_buttons_width_from_icons" data_type="constant" type="int" default="1">
+			If not [code]0[/code], the minimum button width corresponds to the widest of all icons set on those buttons, even if [theme_item buttons_width] is smaller.
```

Comment:
Unless there is use-case for negative width, `buttons_width` being `-1` could be interpreted as using width from icons.

Reply:
These are the possible cases:
- Do not enforce min width, just use the icon for that
- Enforce min width and let the icon enlarge it
- Enforce width and do not let the icon enlarge the buttons - even if there is some overflow. Useful for dense admin panels for example.

In the updated default & editor themes, I set the min width to have a little more of clickable area for the buttons. At the same time, existing themes may rely on the icon autogrow feature and I do not want to break it. So I configured those to use the second case.

Reply:
What I mean is that it's redundant to have both `buttons_width` and `set_min_buttons_width_from_icons`. If you use the latter, the width is useless. Unless it's bigger than button width, in which case width from icons is useless.

We can assume `buttons_width` being negative to have the same effect as `set_min_buttons_width_from_icons` being enabled. Because either you set the width, or let it be determined automatically. This allows to remove `set_min_buttons_width_from_icons`. Unless there is a case I'm missing?

Reply:
I'd like elaboration on this. What KoBeWi is saying makes sense to me, as well.